src/garage/tf/algos/rl2.py

```diff
@@ -0,0 +1,316 @@
+"""RL^2: Fast Reinforcement learning via slow reinforcement learning.
+
+Reference: https://arxiv.org/pdf/1611.02779.pdf.
+"""
+from dowel import logger, tabular
+import numpy as np
+
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.np.algos import RLAlgorithm
+
+
+class RL2(RLAlgorithm):
+    """RL^2.
+
+    Args:
+        inner_algo (garage.np.algos.RLAlgorithm): Inner algorithm.
+        max_path_length (int): Maximum length for trajectories with respect
+            to RL^2. Notice that it is different from the maximum path length
+            for the inner algorithm.
+
+    """
+
+    def __init__(self, inner_algo, max_path_length):
+        self._inner_algo = inner_algo
```

Comment:
what are valid values for `inner_algo`? What is its interface? If we don't have a proper interface for this, you can at least add some checks to make sure the input is sane.

Reply:
Looks like `garage.tf.algos.BatchPolopt` has the proper interface for it. Does it make sense to enforce `inner_algo` to be this type? or better to define a `TfRLAlgorithm`, have `garage.tf.algos.BatchPolopt` inherited from it and enforce `inner_algo` to be `TfRLAlgorithm`?
Basically we want the algo to have `optimize_policy` (After a second check, `log_diagnostics` seems to be deprecated and unnecessary).

Reply:
We don't have a true interface for "works with RL2," and defining one is difficult, so I suggest 2 solutions to this problem (they are not mutually exclusive):

1) `inner_algo` is a string argument with possible values `vpg`, `trpo`, and `ppo` (did I miss any?). Your constructor configures RL2 to use the private interfaces accordingly.
2) `RL2` is not a public class users instantiate -- instead they usually instantiate classes `RL2VPG`, `RL2PPO`, and `RL2TRPO` which implement the respective variants.

Reply:
In TF branch, all `vpg`, `trpo` and `ppo` inherit from `BatchPolopt` which has the only required method `optimize_policy` as its abstract method. In this case, I can enforce `inner_algo` to be a `garage.tf.algos.BatchPolopt`?

Reply:
this is fine with me. i'll refrain from making more comments until i see the example.